aws/resource_aws_ssm_association_test.go

```diff
@@ -40,6 +40,7 @@ func TestAccAWSSSMAssociation_withTargets(t *testing.T) {
 				Config: testAccAWSSSMAssociationBasicConfigWithTargets(name),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAWSSSMAssociationExists("aws_ssm_association.foo"),
+					resource.TestCheckResourceAttr("aws_ssm_association.foo", "targets.#", "2"),
```

Comment:
Do you mind adding a test step here which will add or remove a target and check the no of targets in each step to verify that the update actually works?